### PR TITLE
Fix: add version select

### LINKED ISSUE #200: Add the version selection bar
**Is your feature request related to a problem? Please describe.**
V1 has the section where you can select the version number, V2 doesn't have this feature, making the users to run v2 queries by default unless they manually edit the url. 

**Describe the solution you'd like**
Introduce the select bar like the one in the screenshot below to enable selection of the version

image

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by cwahome**
## Overview

Adds a version selector option
The version selection changes automatically when you select queries from the sidebar.

Improves #205 
Fixes #200 

### Demo

image

## Testing Instructions

* Change the version from v1.0 to beta and notice how the query url changes
* Click on a url in the queries and notice the change in the version selector

### CODE REVIEW

**Comment by jobala** [comment]
@bettirosengugi a while back there was the issue reported by @jthake-msft where for certain screen sizes the http methods were shown to have an ellipsis ie `D...` for `DELETE`. This implementation will re-introduce that behaviour. Please advise.

**Comment by bettirosengugi** [comment]
> @bettirosengugi a while back there was the issue reported by @jthake-msft where for certain screen sizes the http methods were shown to have an ellipsis ie D... for DELETE. This implementation will re-introduce that behaviour. Please advise.

@jobala why don't we have the same issue in classic yet it has the three tabs (Request Options, Version and query tabs) ? May be investigate?

**Comment by bettirosengugi** [comment]
> In the current version of Graph Explorer changing the version in the URL changes the selected version in the version dropdown. I am assuming this is how this feature should behave as well. @bettirosengugi  is this correct?

This is correct to give an example if the version selected is V1, the url for /me should be https://graph.microsoft.com/v1.0/me  and when you change the version to beta the url should change to https://graph.microsoft.com/beta/me 

@cwahome  @jobala

File: src/app/views/query-runner/QueryRunner.tsx [hunk 1]
```diff
@@ -85,10 +80,23 @@ export class QueryRunner extends Component<
     }
   };
 
-  public render() {
-    const { httpMethods } = this.state;
-    const { graphExplorerMode, isLoadingData } = this.props;
+  private handleOnVersionChange = (option?: IDropdownOption) => {
+    const { sampleQuery } = this.props;
+    if (option) {
```
Comment (jobala) [inline src/app/views/query-runner/QueryRunner.tsx]: Rename this to a more descriptive name.

Variables should describe the data they hold.

for example, urlVersion

File: src/app/views/query-runner/QueryRunner.tsx [hunk 2]
```diff
@@ -85,10 +80,23 @@ export class QueryRunner extends Component<
...
+      const urlObject: URL = new URL(sampleQuery.sampleUrl);
+      const requestUrl = urlObject.pathname.substr(5).replace(/\/$/, '');
+      const sampleUrl = `${GRAPH_URL}/${option.text + requestUrl + decodeURI(urlObject.search)}`;
+
+      this.props.actions!.selectQueryVersion(option.text);
```
Comment (jobala) [inline src/app/views/query-runner/QueryRunner.tsx]: This action creator sets the query version. A more appropriate name will be setQueryVersion.

File: src/app/views/sidebar/history/History.tsx [hunk 3]
```diff
@@ -233,8 +233,10 @@ export class History extends Component<IHistoryProps, any> {
   private onRunQuery = (query: IHistoryItem) => {
     const { actions } = this.props;
```
Comment (jobala) [inline src/app/views/sidebar/history/History.tsx]: Unnecessary newline.

**Comment by jobala** [comment]
> This is correct to give an example if the version selected is V1, the url for /me should be https://graph.microsoft.com/v1.0/me and when you change the version to beta the url should change to https://graph.microsoft.com/beta/me

There's also a third type of url version `other`. This is used when a user specifies a version that is neither `v1.0` or `beta`

**Comment by jobala** [comment]
>  why don't we have the same issue in classic yet it has the three tabs (Request Options, Version and query tabs) ? May be investigate?

That's a good question. AFAIK classic uses MWF and `v2` uses Fabric, these are different component providers. It's worth investigating.

**Comment by cwahome** [comment]
> In the current version of Graph Explorer changing the version in the URL changes the selected version in the version dropdown...

Now changes when you change the url in the editor. Good catch @jobala 

> @bettirosengugi a while back there was the issue reported by @jthake-msft where for certain screen sizes the http methods were shown to have an ellipsis ie D... for DELETE. This implementation will re-introduce that behaviour. Please advise.

I have improved the column definitions to reduce the need for the ellipsis in smaller screen sizes

> The above code snippet has been duplicated in various places. It will be wise to abstract it into a function.

I have added a utility function for reuse for this.

> Why do we need a reducer and an action-creator for the version selector?

The version is used and changed at different points of the application.

**Pull request merged**